build/proxies-scenarios.yml

```diff
@@ -96,9 +109,9 @@ steps:
   - ${{ each payload in parameters.payloads }}:
     - ${{ each protocol in parameters.protocols }}:
       # doesn't requiresServerHttps or supportsServerHttps AND doesn't requiresServerHttp2 or supportsServerHttp2
-      - ${{ if and( or( not( protocol.requiresServerHttps), s.supportsServerHttps), or( not( protocol.requiresServerHttp2), s.supportsServerHttp2) ) }}:
+      - ${{ if and( or( not( protocol.requiresServerHttps), s.supportsServerHttps), or( not( protocol.requiresServerHttp2), s.supportsServerHttp2) ), or( not( protocol.requiresDownstreamHttp2), s.supportsDownstreamHttp2) ), or( not( protocol.requiresServerGRPC), s.supportsGRPC) ) }}:
         # For .NET 5.0, only run HTTP-HTTP and H2-H2 benchmarks
-        - ${{ if or( not( eq( s.displayName, 'YARP-net50')), in( protocol.displayName, 'http - http', 'h2 - h2') ) }}:
+        - ${{ if and( not(s.requiresServerGRPC), or( not( eq( s.displayName, 'YARP-net50')), in( protocol.displayName, 'http - http', 'h2 - h2') ) ) }}:
```

Comment:
This new condition, where it's placed, should prevent any `requiresServerGRPC` scenario from running. This line is intended as a special rule for yarp50.

Reply:
I see you are moving it in another "if". However I still don't think this should be on this line. Or please change the comment to explain that it's for both yarp50 and also to exclude grpc ones.

Reply:
Moved that `if` to a different line. `else` would be better here, but I don't know how to use it.